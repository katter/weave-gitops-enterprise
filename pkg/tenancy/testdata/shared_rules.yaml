rbacRules:
  - name: standard-rules
    rules:
    - apiGroups:
        - ''
      resources:
        - 'namespaces'
        - 'pods'
      verbs:
        - 'list'
        - 'get'
tenants:
  - name: foo-tenant
    namespaces:
    - foo-ns
    teamRBAC:
      applyRules: standard-rules
      groupNames:
        - test-group
      rules:
        - apiGroups:
            - ''
          resources:
            - 'namespaces'
            - 'pods'
          verbs:
            - 'create'
            - 'update'
            - 'delete'
