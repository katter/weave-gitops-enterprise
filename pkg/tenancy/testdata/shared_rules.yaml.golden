apiVersion: v1
kind: Namespace
metadata:
  creationTimestamp: null
  labels:
    toolkit.fluxcd.io/tenant: foo-tenant
  name: foo-ns
spec: {}
status: {}
---
apiVersion: v1
kind: ServiceAccount
metadata:
  creationTimestamp: null
  labels:
    toolkit.fluxcd.io/tenant: foo-tenant
  name: foo-tenant
  namespace: foo-ns
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  creationTimestamp: null
  labels:
    toolkit.fluxcd.io/tenant: foo-tenant
  name: foo-tenant
  namespace: foo-ns
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: ClusterRole
  name: cluster-admin
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: User
  name: gotk:foo-ns:reconciler
- kind: ServiceAccount
  name: foo-tenant
  namespace: foo-ns
---
apiVersion: rbac.authorization.k8s.io/v1
kind: Role
metadata:
  creationTimestamp: null
  name: foo-tenant-team-role
  namespace: foo-ns
rules:
- apiGroups:
  - ""
  resources:
  - namespaces
  - pods
  verbs:
  - create
  - update
  - delete
- apiGroups:
  - ""
  resources:
  - namespaces
  - pods
  verbs:
  - list
  - get
---
apiVersion: rbac.authorization.k8s.io/v1
kind: RoleBinding
metadata:
  creationTimestamp: null
  labels:
    toolkit.fluxcd.io/tenant: foo-tenant
  name: foo-tenant-team-rolebinding
  namespace: foo-ns
roleRef:
  apiGroup: rbac.authorization.k8s.io
  kind: Role
  name: foo-tenant-team-role
subjects:
- apiGroup: rbac.authorization.k8s.io
  kind: Group
  name: test-group
---
